File: core-fw-spring/src/main/java/com/firelord/core/ioc/IOCUtils.java
package com.firelord.core.ioc;

public class IOCUtils {
	// #region Construction

	private static IOCUtils m_oInstance;

	private IOCUtils() {

	}

	public synchronized static IOCUtils getInstance() {
		if (m_oInstance == null) {
			m_oInstance = new IOCUtils();
		}
		return m_oInstance;
	}

	// #endregion

	// #region getBean

	public <T> T getBean(Class<T> oBeanClass) {
		return ServiceAwareHelper.getInstance().getApplicationContext().getBean(oBeanClass);
	}

	public Object getBean(String strBeanId) {
		return ServiceAwareHelper.getInstance().getApplicationContext().getBean(strBeanId);
	}

	// #endregion
}

File: cs/core-fw-spring/core/ioc/IOCUtils.cs
using Autofac;
using Autofac.Configuration;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace core_fw.core
{
    /// <summary>
    /// IOCUtils
    /// </summary>
    public class IOCUtils
    {
        #region Fields

        private static readonly IContainer m_oContainer;

        #endregion

        #region Construction

        static IOCUtils()
        {
            var builder = new ContainerBuilder();
            builder.RegisterModule(new ConfigurationSettingsReader("autofac"));
            m_oContainer = builder.Build();
        }

        #endregion

        #region getBean

        /// <summary>
        /// getBean
        /// </summary>
        /// <typeparam name="T">T</typeparam>
        /// <returns>T</returns>
        public static T getBean<T>()
        {
            T oFac = m_oContainer.Resolve<T>();
            return oFac;
        }

        /// <summary>
        /// getBean
        /// </summary>
        /// <typeparam name="T">T</typeparam>
        /// <param name="strName">bean name</param>
        /// <returns>T</returns>
        public static T getBean<T>(String strName)
        {
            T oFac = m_oContainer.ResolveNamed<T>(strName);
            return oFac;
        }

        #endregion
    }
}
